Libraries/Opc.Ua.Security.Certificates/X509Crl/X509Crl.cs

```diff
@@ -0,0 +1,345 @@
+                throw new CryptographicException("Could not verify signature on CRL.");
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Returns true if the certificate is revoked in the CRL.
+        /// </summary>
+        public bool IsRevoked(X509Certificate2 certificate)
+        {
+            if (certificate.IssuerName.Equals(IssuerName))
+            {
+                throw new CryptographicException("Certificate was not created by the CRL Issuer.");
+            }
+            EnsureDecoded();
+            var serialnumber = certificate.GetSerialNumber();
+            foreach (var revokedCert in RevokedCertificates)
+            {
+                if (serialnumber.SequenceEqual<byte>(revokedCert.UserCertificate))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+        #endregion
+
+        #region Private Methods
+        /// <summary>
+        /// Decode the complete CRL.
+        /// </summary>
+        /// <param name="crl">The raw signed CRL</param>
+        internal void Decode(byte[] crl)
+        {
+            // Decode the Tbs and signature
+            m_signature = new X509Signature(crl);
+            // Decode the TbsCertList
+            DecodeCrl(m_signature.Tbs);
+        }
+
+        /// <summary>
+        /// Decode the Tbs of the CRL.
+        /// </summary>
+        /// <param name="tbs">The raw TbsCertList of the CRL.</param>
+        internal void DecodeCrl(byte[] tbs)
+        {
+            try
+            {
+                AsnReader crlReader = new AsnReader(tbs, AsnEncodingRules.DER);
+                var tag = Asn1Tag.Sequence;
+                var seqReader = crlReader?.ReadSequence(tag);
+                if (seqReader != null)
+                {
+                    // Version is OPTIONAL
+                    uint version = 0;
+                    var intTag = new Asn1Tag(UniversalTagNumber.Integer);
+                    var peekTag = seqReader.PeekTag();
+                    if (peekTag == intTag)
+                    {
+                        if (seqReader.TryReadUInt32(out version))
+                        {
+                            if (version != 1)
+                            {
+                                throw new AsnContentException($"The CRL contains an incorrect version {version}");
+                            }
+                        }
+                    }
+
+                    // Signature Algorithm Identifier
+                    var sigReader = seqReader.ReadSequence();
+                    var oid = sigReader.ReadObjectIdentifier();
+                    m_hashAlgorithmName = Oids.GetHashAlgorithmName(oid);
+
+                    // Issuer
+                    m_issuerName = new X500DistinguishedName(seqReader.ReadEncodedValue().ToArray());
+
+                    // thisUpdate
+                    m_thisUpdate = seqReader.ReadUtcTime().UtcDateTime;
```

Comment:
Is reading (decode) really already an update?

Reply:
`thisUpdate` is the name of a data field.  "When was this version of the CRL updated".  and `nextUpdate` is "I promise there will be a newer version by"